### DISCUSSION TITLE
What are 0xcafe, 0xC0FEE, and others seen in every tutorial?

### DISCUSSION THREAD

**Discussion by 0xmaayan** [aptos-labs/aptos-developer-discussions#204] [Questions] [answered]
### Discord user ID

julizamt

### Describe your question in detail.

What are 0xcafe, 0xC0FEE, and others seen in every tutorial? Are they special addresses anyhow or just placeholders?

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by gregnazario** [comment, marked as answer]
These are just placeholders.  When I used to do computer hardware, we often used some fun "hex words" that can be used as an easy to recognize placeholder.

The docs often refer to either:
1. [`0x42` Hitchhikers guide](https://en.wikipedia.org/wiki/42_(number)#The_Hitchhiker's_Guide_to_the_Galaxy)
2. `0xCAFE` or `0xcafe` -> Like a [Cafe](https://en.wikipedia.org/wiki/Coffeehouse)
3. `0xC0FFEE` or `0xC0FEE` -> Like a [cup of Coffee](https://en.wikipedia.org/wiki/Coffee)
4. [`0xDEAD`, `0xBEEF` or `0xDEADBEEF`](https://en.wikipedia.org/wiki/Magic_number_(programming)#DEADBEEF)
5. `0xB0B`
6. `0xA11CE`
7. `0xBAD`

And plenty more https://en.wikipedia.org/wiki/Magic_number_(programming)#Debug_values

**Reply by kshitijc1506** [reply]
Something to note here is that we need to use characters that are valid in the hexadecimal numerical system, which only includes the digits 0-9 and the letters A-F (case-insensitive). 

So if you try and use something like `0xhello` as a placeholder, it wouldn't work.

**Reply by kshitijc1506** [reply]
Here's an [example](https://github.com/aptos-labs/aptos-core/blob/19a3fcb161d38b1a147ddc5be1277f33d14cb16f/aptos-move/move-examples/rewards_pool/simple/sources/tests/rewards_tests.move#L11) of multiple being used in Move unit tests.
